src/solvers/smt2_incremental/smt_commands.def

```diff
@@ -0,0 +1,8 @@
+COMMAND_ID(assert)
+COMMAND_ID(check_sat)
+COMMAND_ID(declare_function)
+COMMAND_ID(define_function)
+COMMAND_ID(exit)
+COMMAND_ID(get_value)
+COMMAND_ID(set_logic)
+COMMAND_ID(set_option)
```

Comment:
Are `(push)` and `(pop)` out of scope for this PR?

Reply:
I had not considered them yet. However I think it will make sense to add them early on. So I will likely add them before I move this PR out of draft status.

Reply:
Would it be possible to add `check-sat-assuming` as well please?

Reply:
Incremental solving should be possible without `check-sat-assuming` right? I'd prefer not to add too many different things up front, so that we can get a MVP working initially and then add incremental improvements to the incremental solving support. There is no reason that more data structures cannot be added as they are needed.

Reply:
I have now added `(push n)` and `(pop n)`.